### PR TITLE
add pipe operator

### PR DESCRIPTION
**Pull request by q60**
> it works but help needed, i'm in love/hate relationships with tree-sitter specifically and parsers in general

#### before

image

#### after

image

### CODE REVIEW

**Comment by eclairevoyant** [comment]
It's a ggod start, but this misses support for the `<|` operator

**Review by oxalica — CHANGES_REQUESTED** [review]
Thanks, I just know that Nix got new experimental operators. It may worth referencing the source PR https://github.com/NixOS/nix/pull/11131 

CI failure is unrelated and is already fixed on `main`. A rebase should fix it.

File: crates/ide/src/ty/infer.rs [hunk 1]
```diff
@@ -250,7 +250,7 @@ impl<'db> InferCtx<'db> {
                         self.unify_var(lhs_ty, rhs_ty);
                         lhs_ty
                     }
-                    BinaryOpKind::Concat => {
+                    BinaryOpKind::Concat | BinaryOpKind::Pipe => {
```
[the remark this answers is not in the document]
Comment (q60) [reply crates/ide/src/ty/infer.rs:253]: i don't quite follow this, need an example... speaking of the other threads, i can make the changes and push them right away
that's quite dumb, but i have basically zero experience in parsers :skull: 
this PR (plus patches in `tree-sitter-nix`, `rnix-parser` and `alejandra`) emerged from my need of using pipe operator w/o errors from the language server and with working formatter. then i showed all of this to friends and they kinda expected me to make PRs to the base repos :skull: :skull: :skull: 
btw i also opened a PR in `tree-sitter-nix`, but considering the status quo of `rnix-parser` and it being a dependency of `alejandra`, i am totally unsure how to work this out :sob: 
sorry for this silly venting in this thread

**Comment by viperML** [comment]
I hate to be that one that writes "any progress on this?", but I need to be that one

**Pull request closed without merging**